File: python/0088.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        if not nums2:return
        a,b=0,0
        while a<m and b<n:
            if nums1[a]<=nums2[b]:
                a+=1
            else:
                nums1[a],nums2[b]=nums2[b],nums1[a]
                j=b
                while j+1<len(nums2) and nums2[j]>nums2[j+1]:
                    nums2[j+1],nums2[j]=nums2[j],nums2[j+1]
                    j+=1
        while 1:
            nums1[a]=nums2[b]
            a+=1
            b+=1
            if b==n:
                break

File: rust/src/bin/0088.rs
struct Solution;

impl Solution {
    pub fn merge(nums1: &mut Vec<i32>, m: i32, nums2: &mut Vec<i32>, n: i32) {
        for i in m..m + n {
            nums1[i as usize] = nums2[(i - m) as usize];
        }
        nums1.sort();
    }
}

fn main() {
    let mut a = vec!(-1, 0, 0, 1);
    let mut b = vec!(-1, -1, 0, 2);
    let m = a.len();
    let n = b.len();
    a.append(&mut vec!(0; b.len()));
    Solution::merge(&mut a, m as i32, &mut b, n as i32);
    println!("{:?}", a);
    println!("{:?}", b);
}